### PR TITLE
EndpointFilter: Testing with Mocks

### PR DESCRIPTION
**Pull request by denisdimitrov**

### CODE REVIEW

File: src/endpointfilter/Endpoint.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package endpointfilter;
+
+/**
+ * Created by clouway on 09.08.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public interface Endpoint {
+    boolean matches(String url);
```
Comment (EmilGeorgiev) [inline src/endpointfilter/Endpoint.java]: documentation

File: src/endpointfilter/StartsWithEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Created by clouway on 09.08.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class StartsWithEndpoint implements Endpoint {
+    @Override
+    public boolean matches(String url){
+        return url.startsWith("www.");
```
Comment (EmilGeorgiev) [inline src/endpointfilter/StartsWithEndpoint.java]: "www." should be pass as parameter of constructor. Now by this way you test state not behavior.  Find articles in internet for State vs Behavior in testing

**Comment by denisdimitrov** [comment]
Everything is fixed.

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**